src/main/java/com/liefery/android/stop_badge/StopBadgeFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.liefery.android.stop_badge;
+
+import android.annotation.SuppressLint;
+import android.graphics.*;
+import android.util.Log;
+import android.util.LruCache;
+
+@SuppressLint( "NewApi" )
+public class StopBadgeFactory {
```

Comment:
I chose singleton to have the cache be accessible from everywhere. Calling `StopBadgeFactory.getBitmap()` from anywhere will always use the same cache, hence the XML <-> Java interoperability.